fuse_variables/src/orientation_3d_stamped.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2018, Locus Robotics
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <fuse_core/uuid.h>
+#include <fuse_variables/orientation_3d_stamped.h>
+
+#include <ceres/local_parameterization.h>
+
+namespace fuse_variables
+{
+
+Orientation3DStamped::Orientation3DStamped(const ros::Time& stamp, const fuse_core::UUID &hardware_id) :
+  hardware_id_(hardware_id),
+  stamp_(stamp),
+  uuid_(fuse_core::uuid::generate(type(), stamp_, hardware_id_))
+{
+}
+
+void Orientation3DStamped::print(std::ostream& stream) const
+{
+  stream << type() << ":\n"
+         << "  uuid: " << uuid() << "\n"
+         << "  hardware id: " << hardware_id() << "\n"
+         << "  stamp: " << stamp() << "\n"
+         << "  size: " << size() << "\n"
+         << "  data:\n"
+         << "  - w: " << w() << "\n"
+         << "  - x: " << x() << "\n"
+         << "  - y: " << y() << "\n"
+         << "  - z: " << z() << "\n";
+}
+
+fuse_core::Variable::UniquePtr Orientation3DStamped::clone() const
+{
+  return Orientation3DStamped::make_unique(*this);
+}
+
+ceres::LocalParameterization* Orientation3DStamped::localParameterization() const
+{
+  ceres::LocalParameterization* local_parameterization = new ceres::QuaternionParameterization();
```

Comment:
Do you need an include for this? Or is it part of `ceres/local_parameterization.h`

Also, you could just `return new ceres::QuaternionParameterization();` ... instead of making the temporary. But I would be willing to bet the compiler gets rid of it anyway.

Reply:
It's inside `local_parameterization.h`. I'll simplify the return.